internal/extensionapi/serverroute_connection.go

```diff
@@ -182,6 +236,19 @@ func (s *ExtensionServer) HandleConnectionCreate(w http.ResponseWriter, r *http.
 		return
 	}
 
+	// Pre-validate WebUI connections before attempting to generate URL
+	if req.Spec.WorkspaceConnectionType == connectionv1alpha1.ConnectionTypeWebUI {
+		if err := s.validateWebUIConnection(namespace, req.Spec.WorkspaceName, logger); err != nil {
+			// Determine appropriate HTTP status code based on error message
+			statusCode := http.StatusBadRequest
```

Comment:
is this right to return bad request if it's not available? It could be internal error, right?